lib/fog/vsphere/models/compute/server.rb

```diff
@@ -319,6 +320,13 @@ def initialize_scsi_controller
             Fog::Compute::Vsphere::SCSIController.new(self.attributes[:scsi_controller])
           end
         end
+
+        def initialize_virtual_machine_summary
+          if attributes[:summary] && attributes[:summary].props
+            self.attributes[:summary] = Fog::Compute::Vsphere::VirtualMachineSummary.new(self.attributes[:summary].props)
```

Comment:
Yep. This will get the attributes and save in another instance variable inside the object
and when calling the getter it will get those attributes and initialize an object for you out of the box.